### PR TITLE
MULE-12826: ParameterModel should be able to point to a ComponentModel

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ElementReference.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.api.meta.model.parameter;
+
+
+/**
+ * Represents a reference to an Extension Configuration for {@link ParameterizedModel}s.
```
Comment (estebanwasinger) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ElementReference.java]: for the next time -> `{@link ParameterizedModel parameterized models}` | `{@link ParameterizedModel ParameterizedModels}`

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterModel.java [hunk 2]
```diff
@@ -93,5 +93,14 @@
   @Override
   Optional<LayoutModel> getLayoutModel();
 
+  /**
+   * @return a {@link List} of {@link ElementReference}s that this {@link ParameterModel} should accept values of.
+   */
+  List<ElementReference> getElementReferences();
+
+  /**
+   * @return A {@link ValueProviderModel} to communicate if the this parameter model is capable to provide {@link Value values}.
```
Comment (estebanwasinger) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterModel.java]: tkm

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterDeclaration.java [hunk 3]
```diff
@@ -38,6 +42,7 @@
   private LayoutModel layoutModel;
   private ParameterRole parameterRole = BEHAVIOUR;
   private ValueProviderModel valueProviderModel;
+  private List<ElementReference> elementReferences = new ArrayList<>();
```
Comment (estebanwasinger) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterDeclaration.java]: `new ArrayList<>()` -> `emptyList()`

Comment (juandesi) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterDeclaration.java]: no se puede, emptyList() no te deja agregar elementos

**Pull request merged**